src/chemistry/geoschem/cesmgc_history_mod.F90

```diff
@@ -0,0 +1,1191 @@
+!                  GEOS-Chem Global Chemical Transport Model                  !
+!------------------------------------------------------------------------------
+!BOP
+!
+! !IROUTINE: Check_HistoryExportsList
+!
+! !DESCRIPTION:
+!\\
+!\\
+! !INTERFACE:
+!
+  SUBROUTINE Check_HistoryExportsList ( am_I_Root, name,  &
+                                        ExportsList, found, RC )
+!
+! !INPUT PARAMETERS:
+!
+    LOGICAL,           INTENT(IN)        :: am_I_Root
+    CHARACTER(LEN=*),  INTENT(IN)        :: name
+    TYPE(HistoryExportsListObj), POINTER :: ExportsList
+!
+! !OUTPUT PARAMETERS:
+!
+    LOGICAL, INTENT(OUT)               :: found
+    INTEGER, INTENT(OUT)               :: RC
+!
+! !REVISION HISTORY:
+!  12 Sep 2017 - E. Lundgren - Initial version
+!  See https://github.com/geoschem/geos-chem for history
+!EOP
+!------------------------------------------------------------------------------
+!BOC
+!
+! !LOCAL VARIABLES:
+!
+    TYPE(HistoryExportObj), POINTER :: current
+
+    __Iam__('Check_HistoryExportsList (cesmgc_history_mod.F90)')
+    RC = GC_SUCCESS
+
+    ! Assume not found
+    found = .False.
+
+    current => ExportsList%head
+    DO WHILE ( ASSOCIATED( current ) )
+       IF ( current%name == name ) THEN
+          found = .TRUE.
+          RETURN
+       ENDIF
+       current => current%next
+    ENDDO
+    current => NULL()
+
+  END SUBROUTINE Check_HistoryExportsList
+!EOC
+!------------------------------------------------------------------------------
+!                  GEOS-Chem Global Chemical Transport Model                  !
+!------------------------------------------------------------------------------
+!BOP
+!
+! !IROUTINE: HistoryExports_SetServices
+!
+! !DESCRIPTION:
+!\\
+!\\
+! !INTERFACE:
+!
+  SUBROUTINE HistoryExports_SetServices( am_I_Root, config_file, &
+                                         HistoryConfig,  RC )
+!
+! !USES:
+!
+    USE cam_history, only: addfld, add_default, horiz_only
+    USE Registry_Params_Mod
+!
+! !INPUT PARAMETERS:
+!
+    LOGICAL,             INTENT(IN)    :: am_I_Root
+    CHARACTER(LEN=*),    INTENT(IN)    :: config_file
+!
+! !INPUT AND OUTPUT PARAMETERS:
+!
+
+!
+! !OUTPUT PARAMETERS:
+!
+    TYPE(HistoryConfigObj), POINTER    :: HistoryConfig  ! History config object
+    INTEGER,             INTENT(OUT)   :: RC
+!
+! !REMARKS:
+!  !
+! !REVISION HISTORY:
+!  01 Sep 2017 - E. Lundgren - Initial version
+!  See https://github.com/geoschem/geos-chem for history
+!EOP
+!------------------------------------------------------------------------------
+!BOC
+!
+! !LOCAL VARIABLES:
+!
+    CHARACTER(LEN=255)                   :: ErrMsg
+    TYPE(HistoryExportObj),      POINTER :: current
+
+    ! ================================================================
+    ! HistoryExports_SetServices begins here
+    ! ================================================================
+
+    ! For MAPL/ESMF error handling (defines Iam and STATUS)
```

Comment:
You should probably scrub mention of MAPL/ESMF in the file other than specifying it is based on the module for that interface.

Reply:
Done.